Variable-sized Coin serialization

sisou opened the pull request:
## What's in this pull request?

This PR makes `Coin` serialize to a varint in transactions and accounts, both over the wire and their internal representations (history store). 

### Reasoning

No transaction will likely ever make use of the full u64 range for its `value`, and especially not for `fee`. So most zero bytes for Coins in most transactions are wasted. This is especially true for signaling transactions, which have 16 zero-bytes to represent 0 value and (mostly) 0 fee. This PR reduces them to two zero bytes.

### Drawbacks

- A transaction could increase in size when adding or increasing the fee, requiring to increase the fee _again_ if targeting a specific fee-per-byte.

### Exceptions

There are two exceptions for backwards-compatibility with PoW transactions:
1. Coins in vesting contract creation data are still serialized with `fixint::be`
2. For determining a transaction's hash, `tx.serialize_content()` serializes `value` and `fee` as before with `fixint::be`

## Pull request checklist

- [x] All tests pass. The project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have resolved all `clippy` and `rustfmt` warnings.

sisou committed:
Fix mempool test to check for fee_per_byte order, instead of absolute fee

sisou committed:
Ensure that transaction hashes stay the same

sisou committed:
Serialize Coin as a variable size int to save space

sisou committed:
Add a newtype around Coin that serializes to big-endian fixint

To de/serialize Coins to big-endian fixint for backwards-compatible vesting contract creation data.

hrxi reviewed (approved):
Looks like a good improvement to me.

sisou committed:
Update test vectors

sisou committed:
Fix encoding of genesis supply into genesis block

Use the mirror function of `u64::from_be_bytes`, which is used when reading it.

hrxi commented:
I think we decided not to do that because it's a breaking change. Is that correct?
